site/src/components/AuthProvider/AuthProvider.tsx

```diff
@@ -0,0 +1,36 @@
+import { useActor, useInterpret } from "@xstate/react"
+import { createContext, FC, PropsWithChildren, useContext } from "react"
+import { authMachine } from "xServices/auth/authXService"
+import { ActorRefFrom } from "xstate"
+
+interface AuthProviderContextValue {
+  authService: ActorRefFrom<typeof authMachine>
+}
+
+const AuthProviderContext = createContext<AuthProviderContextValue | undefined>(
+  undefined,
+)
```

Comment:
Why would this type be potentially undefined? Because the machine hasn't initialized yet? If so, could we do something like:

````
const AuthProviderContext = createContext<AuthProviderContextValue>({
  authService: undefined
})
````

Or whatever the un-initialized value for `authService` would be? This way we can be a little more explicit about what the machine is returning early-state, and we also set ourselves up for the future where the context is returning other methods.

Reply:
It is ok it returns undefined, it should not happen tho but the useAuth hook is doing the type validation already. I see creating empty context values as an anti pattern.  There is a good blog post about this somewhere, I will try to find it.

Reply:
Yes, at somepoint this provider is going to pass values like these instead of machine actors but I thought it would be too much to do right now.

Reply:
The undefined is just a type thing tho since the initialization happens when the component is used on AuthContext.Provider and pass the value props. It also help us to throw an erro if the user is using the context or the hook outside of the Provider.

Reply:
About errors I don't think we need to be too much specific. We make that in the machines but I don't see too much vale tho and it adds significantly work.

Reply:
About the naming, I think I only was able to understand it now 🤦‍♂️ `AuthContext` is definitely better than `AuthProviderContext` for sure.